### PR TITLE
Add sort option to AbstractSearchObject recommendations.

### PR DESCRIPTION
**Pull request by demiankatz**
I thought it might be useful to add the ability to sort search-based recommendations -- e.g. to display "new items" with a `first_indexed desc` sort. I also simplified the comments in searches.ini by making all of the AbstractSearchObject subclasses refer back to CatalogResults for parameter definitions, to reduce the amount of duplicate documentation.

### CODE REVIEW

**Comment by demiankatz** [comment]
While working on this, I also thought it might be useful to build a trait for parsing configurations which could add named argument support similar to PHP's... that would make some configurations easier and more readable. But that's a project for another day!

File: config/vufind/searches.ini [hunk 1]
```diff
@@ -297,19 +298,15 @@ WorkKeys = year
...
 ;       Same as EDSResults, but loaded via AJAX.
-; EPFResults:[GET parameter]:[result limit]:[custom heading]
-;       Display EPF (EBSCO Publication Finder) search results matching the terms
-;       found in the specified GET parameter (default = "lookfor"), limited to a
-;       specified number of matches (default = 5). The optional custom heading
-;       overrides the default for this recommendation.
-; EPFResultsDeferred:[GET parameter]:[result limit]:[custom heading]
+; EPFResults:[GET parameter]:[result limit]:[custom heading]:[filter ini section]:[sort]
+;       Display EPF (EBSCO Publication Finder) search results. See CatalogResults above
+;       for parameter definitions. Note that some may not apply to this search backend.
```
Comment (maccabeelevine) [inline config/vufind/searches.ini]: I like the idea of slimming down the descriptions, but I think we should be clear about which backends support which options, especially with the EBSCO ones where it is so confusing to begin with.  We could still say "See CatalogResults above for parameter definitions" but then say "XYZ are not supported on this backend" or something like that.  I.e. for sort, it is definitely not supported (yet anyway) for the LibGuides recommendations.

And for the [filter ini section] behavior, I didn't design it with EDS and EPF in mind, and so I didn't say in searches.ini that they supported it, but I'm not actually sure either way.

File: module/VuFind/src/VuFind/Recommend/AbstractSearchObject.php [hunk 2]
```diff
@@ -158,9 +166,11 @@ public function init($params, $request)
         }
 
         // Set up the callback to initialize the parameters:
-        $limit = $this->limit;
-        $callback = function ($runner, $params) use ($lookfor, $limit, $typeLabel) {
-            $params->setLimit($limit);
+        $callback = function ($runner, $params) use ($lookfor, $typeLabel) {
+            $params->setLimit($this->limit);
+            if ($this->sort) {
+                $params->setSort($this->sort, true);
```
Comment (maccabeelevine) [inline module/VuFind/src/VuFind/Recommend/AbstractSearchObject.php:172]: What's the intent behind the second param ($force) = true?

**Review by maccabeelevine — CHANGES_REQUESTED** [review]
I like adding the sort option where it works, and I like the documentation simplification in theory, but I'm not sure if it may add some confusion about the details.

File: module/VuFind/src/VuFind/Recommend/AbstractSearchObject.php (hunk 2 above)
Comment (demiankatz) [reply module/VuFind/src/VuFind/Recommend/AbstractSearchObject.php:172]: By default, you can only set sort to a value that's configured in searches.ini, but there are use cases where you might want to use a "hidden" sort value to give recommendations (e.g. you might not have a "sort by latest added to catalog" option in your searches.ini, but you might want to display recommendations based on a `first_indexed desc` sort).

Comment (maccabeelevine) [reply module/VuFind/src/VuFind/Recommend/AbstractSearchObject.php:172]: Ok that definitely makes sense, thanks.

File: config/vufind/searches.ini (hunk 1 above)
Comment (demiankatz) [reply config/vufind/searches.ini]: I've tested the EDS, EPF and Website modules and adjusted the documentation to reflect how things work. I can't test Summon but strongly suspect that my documentation there is correct. I also updated the LibGuides entries to indicate which things won't have any effect. Hopefully that makes this all more clear, though I'm open to further adjustments if you have suggestions.

Comment (maccabeelevine) [reply config/vufind/searches.ini]: Looks great now, and the comments also provide a pretty clear enhancement path if we feel the need.

**Review by maccabeelevine — APPROVED** [review]

**Pull request merged**